b2sdk/account_info/sqlite_account_info.py

```diff
@@ -36,17 +36,39 @@ class SqliteAccountInfo(UrlPoolAccountInfo):
 
     def __init__(self, file_name=None, last_upgrade_to_run=None):
         """
-        If ``file_name`` argument is empty or ``None``, path from ``B2_ACCOUNT_INFO`` environment variable is used. If that is not available, a default of ``~/.b2_account_info`` is used.
```

Comment:
I am reading this docstring over and over and I think apiver doesn't let us change the default location to use `XDG_CONFIG_HOME` in v1. Someone may have written a program that relies on the promise that when the filename was not passed and `B2_ACCOUNT_INFO` was not set, then the db file will land in `~/.b2_account_info`. After running something, the program might clean the environment by removing that file.

If we make this change and `XDG_CONFIG_HOME` is defined, sdk user would not expect the configuration of the end user to be suddenly leaked (as it is not cleaned up anymore), possibly allowing access to the following user.

I think this is a good change and we should introduce it, but that should happen with the release of `from b2sdk.v2 import *`, otherwise we risk introducing a breaking change, while the purpose of apiver is to avoid it unless fixing a bug (which we are not doing here) which would change behavior because the old behavior was just wrong.

Furthermore, we should separate the documentation of the interface from the documentation of the current behavior, so that this can be altered in the future, by mentioning that the user should not define of this algorithm staying exactly the same, but should use `self.filename` to determine the actually determined location of the file in case it needs to be accessed later.

We did not have a compelling reason to release v2, but this is a good one.

@bwbeach @0az do you agree with my viewpoint? Shall we move it to v2?

Reply:
Good catch: I didn't actually think of that case.

Sounds good to me.

Reply:
@0az then would you please adjust the docstring here? We'll hold off merging this until b2sdk 1.1.0 and b2cli 2.0.0 are shipped, then we'll increment apiver to v2 and finally we'll merge your thing (though a final adjustment will be necessary to provide backwards compatibility)

Reply:
Another question for v2: should SqliteAccountInfo support `:memory:` or anonymous (empty string) databases?

https://www.sqlite.org/c3ref/open.html
> If the filename is ":memory:", then a private, temporary in-memory database is created for the connection. This in-memory database will vanish when the database connection is closed. Future versions of SQLite might make use of additional special filenames that begin with the ":" character. It is recommended that when a database filename actually does begin with a ":" character you should prefix the filename with a pathname such as "./" to avoid ambiguity.
> 
> If the filename is an empty string, then a private, temporary on-disk database will be created. This private database will be automatically deleted as soon as the database connection is closed.

Reply:
It looks like that could replace `v` with one line of code. I like less code.